contracts/vending-minter/src/contract.rs

```diff
@@ -74,9 +76,13 @@ pub fn instantiate(
         msg.init_msg.per_address_limit,
```

Comment:
could refactor validation to throw the appropriate error, and save a query `get_max_mintable_tokens`. Also keeps code DRY. is there a reason not to do this?

Reply:
You can't refactor this out because the logic for `display_max_mintable_tokens` is different than `check_dynamic_per_address_limit`. 
In `display_max_mintable_tokens` if per address limit is 100 and max per address limit is 50, it will return 50 so user knows correct mintable tokens. 

In `check_dynamic_per_address_limit` we are not returning the allowed mintable tokens count, but rather returning true or false whether it passes the validation criteria.